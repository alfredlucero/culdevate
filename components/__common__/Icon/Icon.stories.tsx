import React from "react";
import { storiesOf } from "@storybook/react";
import StoryFrame from "../StoryFrame";
import Icon, { CuldevateIcon } from "./index";

const allCuldevateIcons: CuldevateIcon[] = [
  "suitcase",
  "school",
  "award",
  "target",
  "pen",
  "checkDouble",
  "userFriends",
  "trophy",
  "levelUp",
];

storiesOf("Icon", module).add("All Icon Types", () => (
  <StoryFrame>
    {allCuldevateIcons.map((culdevateIcon, idx) => (
      <div key={idx}>
        <Icon icon={culdevateIcon} />
      </div>
    ))}
  </StoryFrame>
));
