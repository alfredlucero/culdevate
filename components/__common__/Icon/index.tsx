import React from "react";
import styled, { css } from "../../../@types/styled-components/index";
import { IconDefinition } from "@fortawesome/fontawesome-svg-core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faSuitcase,
  faGraduationCap,
  faAward,
  faBullseye,
  faPenAlt,
  faCheckDouble,
  faUserFriends,
  faTrophy,
  faLevelUpAlt,
} from "@fortawesome/free-solid-svg-icons";
import { culdevateThemes } from "../../defaultTheme";

export type CuldevateIcon =
  | "suitcase"
  | "school"
  | "award"
  | "target"
  | "pen"
  | "checkDouble"
  | "userFriends"
  | "trophy"
  | "levelUp";

type IconMap = { [key in CuldevateIcon]: IconDefinition };

const culdevateIconToFontAwesomeIconMap: IconMap = {
  suitcase: faSuitcase, // Used for Work Recap
  school: faGraduationCap, // Used for Education Recap
  award: faAward, // Used for Awards Recap
  target: faBullseye, // Used for Objectives Recap
  pen: faPenAlt, // Used for Proof Recap
  checkDouble: faCheckDouble, // Used for Other Recap
  userFriends: faUserFriends, // Used for Extracurricular Recap
  trophy: faTrophy, // Used for Achievements Recap
  levelUp: faLevelUpAlt, // Used for Skills Recap
};

interface IconProps {
  icon: CuldevateIcon;
}

const Icon: React.FC<IconProps> = ({ icon, ...rest }) => {
  const mappedIcon = culdevateIconToFontAwesomeIconMap[icon];
  return <FontAwesomeIcon icon={mappedIcon} {...rest} />;
};

export const StyledRoundIcon = styled.div`
  height: 50px;
  width: 50px;
  border-radius: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 2rem;

  ${({ theme }) => css`
    background-color: ${theme.semanticColors.iconBgColor};
    color: ${theme.semanticColors.iconColor};
  `}
`;

StyledRoundIcon.defaultProps = {
  theme: culdevateThemes.light,
};

export default Icon;
