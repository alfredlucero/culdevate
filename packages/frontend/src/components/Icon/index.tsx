import React from "react";
import cn from "classnames";
import { FontAwesomeIcon, FontAwesomeIconProps } from "@fortawesome/react-fontawesome";
import { IconDefinition } from "@fortawesome/fontawesome-svg-core";
import {
  faListAlt,
  faSuitcase,
  faGraduationCap,
  faAward,
  faBook,
  faCertificate,
  faProjectDiagram,
  faUsers,
  faAddressCard,
  faHighlighter,
  faSquare,
  faTimes,
  faTrashAlt,
  faPlus,
  faFan,
  faPencilAlt,
  faExclamationTriangle,
  faChevronLeft,
} from "@fortawesome/free-solid-svg-icons";
import { CommonProps } from "../commonProps";

export interface IconProps extends CommonProps, Omit<FontAwesomeIconProps, "icon" | "size"> {
  variant: IconVariant;
  size: IconSize;
  id?: string;
  onClick?: (e: React.MouseEvent) => void;
}

export type IconVariant =
  | "allRecaps"
  | "work"
  | "education"
  | "accomplishments"
  | "publications"
  | "skills"
  | "projects"
  | "organizations"
  | "references"
  | "other"
  | "bulletpoint"
  | "x"
  | "trash"
  | "plus"
  | "loadingFan"
  | "editPencil"
  | "exclamationTriangle"
  | "chevronLeft";

export type IconSize = "small" | "medium" | "large" | "xlarge";

const iconVariantToFontAwesomeMap: IconVariantToFontAwesomeMap = {
  allRecaps: faListAlt,
  work: faSuitcase,
  education: faGraduationCap,
  accomplishments: faAward,
  publications: faBook,
  skills: faCertificate,
  projects: faProjectDiagram,
  organizations: faUsers,
  references: faAddressCard,
  other: faHighlighter,
  bulletpoint: faSquare,
  x: faTimes,
  trash: faTrashAlt,
  plus: faPlus,
  loadingFan: faFan,
  editPencil: faPencilAlt,
  exclamationTriangle: faExclamationTriangle,
  chevronLeft: faChevronLeft,
};

type IconVariantToFontAwesomeMap = {
  [iconVariant in IconVariant]: IconDefinition;
};

const Icon: React.FC<IconProps> = ({
  variant,
  size,
  onClick = () => {},
  id = "",
  testId = "",
  className = "",
  ...restOfIconProps
}) => {
  const icon = iconVariantToFontAwesomeMap[variant];

  return (
    <div
      onClick={onClick}
      className={cn(
        "font-sans",
        {
          "text-2xl": size === "small",
          "text-3xl": size === "medium",
          "text-4xl": size === "large",
          "text-6xl": size === "xlarge",
        },
        className,
      )}
      id={id}
      {...(testId !== "" ? { "data-testid": testId } : {})}
    >
      <FontAwesomeIcon icon={icon} {...restOfIconProps} />
    </div>
  );
};

export default Icon;
