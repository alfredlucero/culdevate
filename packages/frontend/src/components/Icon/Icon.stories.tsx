import React from "react";
import { storiesOf } from "@storybook/react";
import Icon, { IconVariant } from "./index";

const onClick = () => {
  console.log("Icon clicked!");
};

const iconVariants: IconVariant[] = [
  "work",
  "education",
  "accomplishments",
  "publications",
  "skills",
  "projects",
  "organizations",
  "references",
  "other",
  "bulletpoint",
  "x",
  "trash",
  "plus",
];

storiesOf("Common/Icon", module)
  .add("All Small Icons", () => (
    <div>
      {iconVariants.map((iconVariant, idx) => {
        return (
          <div key={idx}>
            <p>Icon Variant: {iconVariant}</p>
            <Icon variant={iconVariant} size="small" onClick={onClick} />
          </div>
        );
      })}
    </div>
  ))
  .add("All Medium Icons", () => (
    <div>
      {iconVariants.map((iconVariant, idx) => {
        return (
          <div key={idx}>
            <p>Icon Variant: {iconVariant}</p>
            <Icon variant={iconVariant} size="medium" onClick={onClick} />
          </div>
        );
      })}
    </div>
  ))
  .add("All Large Icons", () => (
    <div>
      {iconVariants.map((iconVariant, idx) => {
        return (
          <div key={idx}>
            <p>Icon Variant: {iconVariant}</p>
            <Icon variant={iconVariant} size="large" onClick={onClick} />
          </div>
        );
      })}
    </div>
  ));
