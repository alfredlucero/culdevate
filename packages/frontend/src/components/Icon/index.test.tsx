import React from "react";
import { render } from "@testing-library/react";
import Icon, { IconVariant } from "./index";

const iconVariants: IconVariant[] = [
  "work",
  "education",
  "accomplishments",
  "publications",
  "skills",
  "projects",
  "organizations",
  "references",
  "other",
  "bulletpoint",
  "x",
  "trash",
  "plus",
  "loadingFan",
  "editPencil",
];

describe("<Icon />", () => {
  const iconTestId = "iconTestId";

  test("should render all icon variants without error", () => {
    const { container } = render(
      <div>
        {iconVariants.map((iconVariant, idx) => {
          return (
            <div key={idx}>
              <p>Icon Variant: {iconVariant}</p>
              <Icon variant={iconVariant} size="small" testId={iconTestId} className="extra-icon-class" />
            </div>
          );
        })}
      </div>,
    );

    expect(container).toMatchSnapshot();
  });
});
