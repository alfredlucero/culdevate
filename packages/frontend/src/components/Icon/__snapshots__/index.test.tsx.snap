// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render all icon variants without error 1`] = `
<div>
  <div>
    <div>
      <p>
        Icon Variant: 
        allRecaps
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-list-alt fa-w-16 "
          data-icon="list-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zM128 120c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm0 96c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm0 96c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm288-136v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12zm0 96v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12zm0 96v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        work
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-suitcase fa-w-16 "
          data-icon="suitcase"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M128 480h256V80c0-26.5-21.5-48-48-48H176c-26.5 0-48 21.5-48 48v400zm64-384h128v32H192V96zm320 80v256c0 26.5-21.5 48-48 48h-48V128h48c26.5 0 48 21.5 48 48zM96 480H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48h48v352z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        education
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-graduation-cap fa-w-20 "
          data-icon="graduation-cap"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M622.34 153.2L343.4 67.5c-15.2-4.67-31.6-4.67-46.79 0L17.66 153.2c-23.54 7.23-23.54 38.36 0 45.59l48.63 14.94c-10.67 13.19-17.23 29.28-17.88 46.9C38.78 266.15 32 276.11 32 288c0 10.78 5.68 19.85 13.86 25.65L20.33 428.53C18.11 438.52 25.71 448 35.94 448h56.11c10.24 0 17.84-9.48 15.62-19.47L82.14 313.65C90.32 307.85 96 298.78 96 288c0-11.57-6.47-21.25-15.66-26.87.76-15.02 8.44-28.3 20.69-36.72L296.6 284.5c9.06 2.78 26.44 6.25 46.79 0l278.95-85.7c23.55-7.24 23.55-38.36 0-45.6zM352.79 315.09c-28.53 8.76-52.84 3.92-65.59 0l-145.02-44.55L128 384c0 35.35 85.96 64 192 64s192-28.65 192-64l-14.18-113.47-145.03 44.56z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        accomplishments
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-award fa-w-12 "
          data-icon="award"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M97.12 362.63c-8.69-8.69-4.16-6.24-25.12-11.85-9.51-2.55-17.87-7.45-25.43-13.32L1.2 448.7c-4.39 10.77 3.81 22.47 15.43 22.03l52.69-2.01L105.56 507c8 8.44 22.04 5.81 26.43-4.96l52.05-127.62c-10.84 6.04-22.87 9.58-35.31 9.58-19.5 0-37.82-7.59-51.61-21.37zM382.8 448.7l-45.37-111.24c-7.56 5.88-15.92 10.77-25.43 13.32-21.07 5.64-16.45 3.18-25.12 11.85-13.79 13.78-32.12 21.37-51.62 21.37-12.44 0-24.47-3.55-35.31-9.58L252 502.04c4.39 10.77 18.44 13.4 26.43 4.96l36.25-38.28 52.69 2.01c11.62.44 19.82-11.27 15.43-22.03zM263 340c15.28-15.55 17.03-14.21 38.79-20.14 13.89-3.79 24.75-14.84 28.47-28.98 7.48-28.4 5.54-24.97 25.95-45.75 10.17-10.35 14.14-25.44 10.42-39.58-7.47-28.38-7.48-24.42 0-52.83 3.72-14.14-.25-29.23-10.42-39.58-20.41-20.78-18.47-17.36-25.95-45.75-3.72-14.14-14.58-25.19-28.47-28.98-27.88-7.61-24.52-5.62-44.95-26.41-10.17-10.35-25-14.4-38.89-10.61-27.87 7.6-23.98 7.61-51.9 0-13.89-3.79-28.72.25-38.89 10.61-20.41 20.78-17.05 18.8-44.94 26.41-13.89 3.79-24.75 14.84-28.47 28.98-7.47 28.39-5.54 24.97-25.95 45.75-10.17 10.35-14.15 25.44-10.42 39.58 7.47 28.36 7.48 24.4 0 52.82-3.72 14.14.25 29.23 10.42 39.59 20.41 20.78 18.47 17.35 25.95 45.75 3.72 14.14 14.58 25.19 28.47 28.98C104.6 325.96 106.27 325 121 340c13.23 13.47 33.84 15.88 49.74 5.82a39.676 39.676 0 0 1 42.53 0c15.89 10.06 36.5 7.65 49.73-5.82zM97.66 175.96c0-53.03 42.24-96.02 94.34-96.02s94.34 42.99 94.34 96.02-42.24 96.02-94.34 96.02-94.34-42.99-94.34-96.02z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        publications
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-book fa-w-14 "
          data-icon="book"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        skills
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-certificate fa-w-16 "
          data-icon="certificate"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M458.622 255.92l45.985-45.005c13.708-12.977 7.316-36.039-10.664-40.339l-62.65-15.99 17.661-62.015c4.991-17.838-11.829-34.663-29.661-29.671l-61.994 17.667-15.984-62.671C337.085.197 313.765-6.276 300.99 7.228L256 53.57 211.011 7.229c-12.63-13.351-36.047-7.234-40.325 10.668l-15.984 62.671-61.995-17.667C74.87 57.907 58.056 74.738 63.046 92.572l17.661 62.015-62.65 15.99C.069 174.878-6.31 197.944 7.392 210.915l45.985 45.005-45.985 45.004c-13.708 12.977-7.316 36.039 10.664 40.339l62.65 15.99-17.661 62.015c-4.991 17.838 11.829 34.663 29.661 29.671l61.994-17.667 15.984 62.671c4.439 18.575 27.696 24.018 40.325 10.668L256 458.61l44.989 46.001c12.5 13.488 35.987 7.486 40.325-10.668l15.984-62.671 61.994 17.667c17.836 4.994 34.651-11.837 29.661-29.671l-17.661-62.015 62.65-15.99c17.987-4.302 24.366-27.367 10.664-40.339l-45.984-45.004z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        projects
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-project-diagram fa-w-20 "
          data-icon="project-diagram"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 320H256c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V352c0-17.67-14.33-32-32-32zM192 32c0-17.67-14.33-32-32-32H32C14.33 0 0 14.33 0 32v128c0 17.67 14.33 32 32 32h95.72l73.16 128.04C211.98 300.98 232.4 288 256 288h.28L192 175.51V128h224V64H192V32zM608 0H480c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        organizations
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-users fa-w-20 "
          data-icon="users"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        references
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-address-card fa-w-18 "
          data-icon="address-card"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M528 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h480c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-352 96c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm112 236.8c0 10.6-10 19.2-22.4 19.2H86.4C74 384 64 375.4 64 364.8v-19.2c0-31.8 30.1-57.6 67.2-57.6h5c12.3 5.1 25.7 8 39.8 8s27.6-2.9 39.8-8h5c37.1 0 67.2 25.8 67.2 57.6v19.2zM512 312c0 4.4-3.6 8-8 8H360c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h144c4.4 0 8 3.6 8 8v16zm0-64c0 4.4-3.6 8-8 8H360c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h144c4.4 0 8 3.6 8 8v16zm0-64c0 4.4-3.6 8-8 8H360c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h144c4.4 0 8 3.6 8 8v16z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        other
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-highlighter fa-w-17 "
          data-icon="highlighter"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 544 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 479.98L99.92 512l35.45-35.45-67.04-67.04L0 479.98zm124.61-240.01a36.592 36.592 0 0 0-10.79 38.1l13.05 42.83-50.93 50.94 96.23 96.23 50.86-50.86 42.74 13.08c13.73 4.2 28.65-.01 38.15-10.78l35.55-41.64-173.34-173.34-41.52 35.44zm403.31-160.7l-63.2-63.2c-20.49-20.49-53.38-21.52-75.12-2.35L190.55 183.68l169.77 169.78L530.27 154.4c19.18-21.74 18.15-54.63-2.35-75.13z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        bulletpoint
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-square fa-w-14 "
          data-icon="square"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        x
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-times fa-w-11 "
          data-icon="times"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 352 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        trash
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-trash-alt fa-w-14 "
          data-icon="trash-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        plus
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus fa-w-14 "
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        loadingFan
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-fan fa-w-16 "
          data-icon="fan"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M352.57 128c-28.09 0-54.09 4.52-77.06 12.86l12.41-123.11C289 7.31 279.81-1.18 269.33.13 189.63 10.13 128 77.64 128 159.43c0 28.09 4.52 54.09 12.86 77.06L17.75 224.08C7.31 223-1.18 232.19.13 242.67c10 79.7 77.51 141.33 159.3 141.33 28.09 0 54.09-4.52 77.06-12.86l-12.41 123.11c-1.05 10.43 8.11 18.93 18.59 17.62 79.7-10 141.33-77.51 141.33-159.3 0-28.09-4.52-54.09-12.86-77.06l123.11 12.41c10.44 1.05 18.93-8.11 17.62-18.59-10-79.7-77.51-141.33-159.3-141.33zM256 288a32 32 0 1 1 32-32 32 32 0 0 1-32 32z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        editPencil
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-pencil-alt fa-w-16 "
          data-icon="pencil-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        exclamationTriangle
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
          data-icon="exclamation-triangle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <p>
        Icon Variant: 
        chevronLeft
      </p>
      <div
        class="font-sans text-2xl extra-icon-class"
        data-testid="iconTestId"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-left fa-w-10 "
          data-icon="chevron-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
